File: strategy/ducks/Dance/Waltz.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace ducks.Dance
{
	public class Waltz : IDanceBehaviuor
	{
		public Waltz(StringBuilder stringBuilder) => _stringBuilder = stringBuilder;

		public void Dance() => _stringBuilder?.Append("waltz");

		private readonly StringBuilder _stringBuilder;
	}
}

File: strategy_java/src/ducks/strategies/Waltz.java
package ducks.strategies;

import ducks.base.IDanceBehaviour;

public class Waltz implements IDanceBehaviour {
    public Waltz(StringBuilder sb) {
        _sb = sb;
    }

    @Override
    public void Dance() {
        _sb.append("waltz");
    }

    private final StringBuilder _sb;
}
